### PR TITLE
Remi- Database services

### PR DESCRIPTION
**Pull request by Remi-dee**
# This PR Closes
- #197  ,
- #198,
-  #199
- #201  

# Does this PR solve any bug?
Resolved `undefined error` from page.js

# What did you work on?
Created a function that add session data to firestore
Created a function that retrieve user details on collaboration
Created a function that add code editor data to firestore

## Created Session
add editor 1

## Updated Session on same ID
Hey

### CODE REVIEW

File: development/src/composables/dbService.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { appFirestore, appAuth, app } from "./firebaseConfig/config";
```
Comment (TaylorDJones11) [inline development/src/composables/dbService.js:1]: Isn't app and appAuth the same thing?

**Review by TaylorDJones11 — CHANGES_REQUESTED** [review]
Just a minor question at the commented code. Otherwise it looks good. Great job Remi!

File: development/src/composables/dbService.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { setDoc, doc, addDoc, updateDoc, getDoc } from "firebase/firestore";
+
+async function addSession(userSessionData) {
+  const { sessionName, peerId, language, userName, peerName } = userSessionData;
+  if (!sessionName || !peerId || !language || !userName || !peerName) {
+    throw new Error("Kindly provide all fields");
+  }
+
+  const user = appAuth.currentUser;
```
Comment (annonymousauthority) [inline development/src/composables/dbService.js:10]: This won't work

File: development/src/composables/dbService.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  if (user == null) {
```
Comment (annonymousauthority) [inline development/src/composables/dbService.js:11]: It'll for some reason claim the user is undefined, even if it's signed in.

**Comment by Remi-dee** [comment]
> Just a minor question at the commented code. Otherwise it looks good. Great job Remi!

Thanks Taylor!

File: development/src/composables/dbService.js (hunk 1 above)
Comment (Remi-dee) [reply development/src/composables/dbService.js]: Yea great spot out
I’ve updated that now.

File: development/src/composables/dbService.js (hunk 3 above)
Comment (annonymousauthority) [reply development/src/composables/dbService.js:11]: Keep the code, but debug it when it doesn't feel like it's working during test

File: development/src/composables/dbService.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+
+async function addCollabCodeEditor(codeEditorData) {
+  const { editorCode, peerId } = codeEditorData;
+  if (!editorCode || !peerId) {
+    throw new Error("Kindly insert text");
+  }
+  const user = appAuth.currentUser;
+  console.log(user);
```
Comment (annonymousauthority) [inline development/src/composables/dbService.js]: Please remove

File: development/src/composables/dbService.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+  const coders = doc(appFirestore, `CODERS/${user.uid}`);
+  const session = doc(coders, `SESSION/${peerId}`);
+  try {
+    const codeData = {
+      codes: editorCode,
+    };
+
+    const docRef = await setDoc(session, codeData, { merge: true });
+    console.log("code editor data added");
```
Comment (annonymousauthority) [inline development/src/composables/dbService.js:75]: Use proper message reporting, users don't open consoles to check for updates

**Review by annonymousauthority — CHANGES_REQUESTED** [review]
This changes are important

File: development/src/composables/dbService.js (hunk 3 above)
Comment (Remi-dee) [reply development/src/composables/dbService.js:11]: Alright.

**Review by annonymousauthority — APPROVED** [review]
Cautionary Merge

**Review by TaylorDJones11 — DISMISSED** [review]

**Review by TaylorDJones11 — DISMISSED** [review]

**Review by annonymousauthority — APPROVED** [review]
Cautionary Merge

**Review by annonymousauthority — APPROVED** [review]
Cautionary Merge

**Review by 7malikk — APPROVED** [review]

**Pull request merged**